configs/components/rubygem-ffi.rb

```diff
@@ -65,8 +65,11 @@
   pkg.environment "CPATH", "/opt/csw/lib/libffi-3.2.1/include" if platform.name =~ /solaris-11/
   pkg.environment "MAKE", platform[:make] if platform.is_solaris?
 
-  if platform.is_cross_compiled_linux? || platform.is_solaris?
-    pkg.environment "PATH", "/opt/pl-build-tools/bin:/opt/csw/bin:$(PATH)"
+  if settings[:runtime_project] == 'agent-runtime-main'
+    if platform.is_cross_compiled_linux? || platform.is_solaris?
```

Comment:
Due to the order of these `if`'s, we no longer override `PATH` when compiling `agent-runtime-7.x` on cross compiled linux or solaris:

```
    Component 'rubygem-ffi'
        Field: environment
        --------------------
        - GEM_HOME=/opt/puppetlabs/puppet/lib/ruby/gems/3.2.0 GEM_PATH=/opt/puppetlabs/puppet/lib/ruby/gems/3.2.0 RUBYLIB=/opt/puppetlabs/puppet/lib/ruby/vendor_ruby:$(RUBYLIB) MAKE=/opt/csw/bin/gmake PATH=/opt/pl-build-tools/bin:/opt/csw/bin:$(PATH)
        + GEM_HOME=/opt/puppetlabs/puppet/lib/ruby/gems/3.2.0 GEM_PATH=/opt/puppetlabs/puppet/lib/ruby/gems/3.2.0 RUBYLIB=/opt/puppetlabs/puppet/lib/ruby/vendor_ruby:$(RUBYLIB) MAKE=/opt/csw/bin/gmake
```

Maybe something like?

```suggestion
    if platform.is_cross_compiled_linux?
      pkg.environment "PATH", "/opt/pl-build-tools/bin:$(PATH)"
    elsif platform.is_solaris?
      if settings[:runtime_project] == 'agent-runtime-main'
        pkg.environment "PATH", "/opt/csw/bin:/opt/pl-build-tools/bin:$(PATH)"
      else
        pkg.environment "PATH", "/opt/pl-build-tools/bin:/opt/csw/bin:$(PATH)"
      end  
    end  
```